### DISCUSSION TITLE
Add Custom Field

### DISCUSSION THREAD

**Discussion by adnanawan12** [payloadcms/payload#1133] [Feature Requests & Ideas]
Is there a way to add custom field instead of using available field types? We want to use html field with.  Draft js editor. Can we achieve this?

If this is not possible then can we replace slate js editor with draft js editor?

**Comment by DanRibbens** [comment]
Hey  @adnanawan12!

To use a different field UI you can set a field to `type: 'richtext'` and your own  `admin.components.Field` can be the react component for the UI you wish to use.

In that way you get all the api and database schema for your custom field along with the custom UI you want to build.

Does this make sense?

Thanks,
Dan

**Reply by adnanawan12** [reply]
Hey. @DanRibbens !

Thanks for your quick reply. I will give it a go :)

**Reply by adnanawan12** [reply]
@DanRibbens 

I am able to create custom control as suggested by you.

Now I am using suneditor-react into my custom control. Whenever I use editor into my control then I get the following error

import blockquote from './command/blockquote';
^^^^^^

SyntaxError: Cannot use import statement outside a module

**Reply by DanRibbens** [reply]
Yes, that is a configuration issue. Add `"type": "module"` to your package.json. You might have to adjust compilerOptions in tsconfig if you're using typescript too.
